hi-ml-multimodal/src/health_multimodal/image/model/model.py

```diff
@@ -75,7 +77,23 @@ class ImageModelOutput():
     projected_patch_embeddings: torch.Tensor
 
 
-class ImageModel(nn.Module):
+class BaseImageModel(nn.Module, ABC):
+    """ An abstract class defining methods required by the ImageInferenceEngine """
+
+    def __init__(self, **kwargs: Any):
+        super().__init__()
+        assert hasattr(self, "training")
+
+    @abstractmethod
+    def forward(self, x: torch.Tensor) -> ImageModelOutput:
```

Comment:
Will this be compatible with a child class taking multiple images?

Also applies to the other method below.

Reply:
I guess a solution would be using `*args` and `**kwargs`, although mypy will probably complain. I suppose there are more elegant solutions, though.

Reply:
I've pushed a possible solution where we use an `ImageModelInput` class, a child class that takes >1 image could then subclass this to add new inputs. This is more explicit than using `*args/**kwargs` but it feels a bit clunky.

Reply:
How about just removing the method? Then children will have freedom to implement their `forward`s however they want.

Reply:
If the forward doesn't return something with at least some of the fields in `ImageModelOutput`, our inference engine methods will fail later.

Reply:
Sure, but the children classes could take care of that.

Reply:
Anyway, this is not terribly important and could be changed in a future PR.

Reply:
Actually, I've changed my mind. Looking at the code, this breaks the current API of the model, which is not great, and passing some custom class to the model instead of just a tensor is not very friendly or PyTorch-standard. So I think it might be better to use the *args syntax or simply delegate the creation of the `forward` method to the children classes.

Reply:
Sounds good!